apps/authentication/views.py

```diff
@@ -83,9 +85,12 @@ def register(request):
 
                 # Create the registration token
                 token = uuid.uuid4().hex
-                rt = RegisterToken(user=user, email=email.email, token=token)
-                rt.save()
-
+                try:
+                    rt = RegisterToken(user=user, email=email.email, token=token)
+                    rt.save()
+                    log.info('Successfully registered token for %s' % request.user)
+                except:
+                    log.error('Failed to register token for %s' % request.user)
```

Comment:
Like creating new tokens until one works you mean? I might be asking stupid questions here, but wouldn't putting that in the `create()`-function make the token different from the one in the program that creates a RegisterToken object? Not sure if that would matter at all though. Also, if the token is changed upon creating the object anyways, would it even need to be passed in as an argument. 

Having the creation of the token in a loop running until it passes seems easier, but that might be a bad way of doing it as it would have to do that every time a RegisterToken is created.